### PR TITLE
mongo_db

### PR DESCRIPTION
**Pull request by michal24509**
create Mongo DB Client

### CODE REVIEW

File: db/mongo_db/mongo_db_client.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @staticmethod
+    def get_connection_string_db(_configuration):
+        db_user_name = _configuration.get('db_user_name', None)
+        if db_user_name is None:
+            return f'mongodb://localhost:27017/'
+        db_password = _configuration.get('db_password')
+        db_atlas_cluster = _configuration.get('db_atlas_cluster')
+        return f'mongodb+srv://{db_user_name}:{db_password}@{db_atlas_cluster}/'
+
+    def get(self, keys: list):
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:21]: you implement the get and set in another PR ?

**Review by hadassaCh — CHANGES_REQUESTED** [review]

File: db/mongo_db/mongo_db_client.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+class MongoDBClient(DBClient):
+    def __init__(self, _name, _configuration):
+        super(MongoDBClient, self).__init__(_name=_name, _configuration=_configuration)
+        mongo_client = MongoClient(self.get_connection_string_db(_configuration=_configuration))
+        self.client = mongo_client[_name]
+
+    @staticmethod
+    def get_connection_string_db(_configuration):
+        db_user_name = _configuration.get('db_user_name', None)
+        if db_user_name is None:
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:15]: Add use_local_db variable in configuration to detect which connection string db to return, insteas use `db_user_name`

File: db/mongo_db/mongo_db_client.py (hunk 1 above)
Comment (michal24509) [reply db/mongo_db/mongo_db_client.py:21]: yes I will  do it after create table

File: db/mongo_db/mongo_db_client.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        mongo_client = MongoClient(self.get_connection_string_db(_configuration=_configuration))
+        self._client = mongo_client[_name]
+
+    def get_connection_string_db(self, _configuration):
+        use_local_db = _configuration.get('use_local_db', False)
+        if use_local_db:
+            return f'mongodb://localhost:27017/'
+
+        db_user_name = _configuration.get('db_user_name')
+        env_variable = _configuration.get('db_password')
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:21]: The env_variable need retrieved before using MongoDBClient.
in configuration is sended the password.

File: db/mongo_db/mongo_db_client.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        pass
+
+
+if __name__ == '__main__':
+    db_config = {
+        'db_atlas_cluster': None,
+        'db_password': None,
+        'db_user_name': None,
+    }
+    mongo = MongoDBClient(_name='test_db', _configuration=db_config)
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:47]: update the main

**Review by hadassaCh — CHANGES_REQUESTED** [review]

File: db/mongo_db/mongo_db_client.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        db_password = _configuration.get('db_password')
+        db_atlas_cluster = _configuration.get('db_atlas_cluster')
+        return f'mongodb+srv://{db_user_name}:{db_password}@{db_atlas_cluster}/'
+
+    def get(self, key, _table_name):
+        values = list(self._client[_table_name].distinct(key))
+        return values
+
+    def get_table_by_name(self, _table_name):
+        self._client.get_collection(_table_name)
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:30]: what this function does ? it was not return anything

**Review by hadassaCh — CHANGES_REQUESTED** [review]

File: db/mongo_db/mongo_db_client.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import json
+from pymongo import MongoClient
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:2]: You need update the requiement.txt file ?

File: db/mongo_db/mongo_db_client.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        return f'mongodb+srv://{db_user_name}:{db_password}@{db_atlas_cluster}/'
+
+    def get(self, key, _table_name):
+        values = list(self._client[_table_name].distinct(key))
+        return values
+
+    def get_table_by_name(self, _table_name):
+        self._client.get_collection(_table_name)
+
+    def set(self, key: str, value: dict, _table_name) -> bool:
```
[the remark this answers is not in the document]
Comment (hadassaCh) [reply db/mongo_db/mongo_db_client.py:29]: you are not update the change

**Review by hadassaCh — CHANGES_REQUESTED** [review]

File: db/mongo_db/mongo_db_client.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        db_user_name = _configuration.get('db_user_name')
+        db_password = _configuration.get('db_password')
+        db_atlas_cluster = _configuration.get('db_atlas_cluster')
+        return f'mongodb+srv://{db_user_name}:{db_password}@{db_atlas_cluster}/'
+
+    def get(self, _key, _table_name):
+        values = list(self._client[_table_name].distinct(_key))
+        return values
+
+    def get_table_by_name(self, _table_name):
```
Comment (hadassaCh) [inline db/mongo_db/mongo_db_client.py:26]: This function already exist in line 33

**Review by hadassaCh — CHANGES_REQUESTED** [review]

**Pull request merged**